.github/workflows/e2e.yaml

```diff
@@ -0,0 +1,39 @@
+name: e2e-mock-provider-tests
+on: 
+  workflow_dispatch:
+    inputs:
+      registry:
+        description: 'Registry'
+        required: true
+        default: 'gcr.io/k8s-staging-csi-secrets-store'
+      driverImage:
+        description: 'Secret Store CSI driver image'
+        required: true
+        default: 'driver'
+      driverImageVersion:
+        description: 'Secret Store CSI driver image version'
+        required: true
+        default: 'v0.3.0'
+jobs:
+  e2e-test:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          submodules: true
+          fetch-depth: 0  
+      - name: Setup BATS
+        # pinning to the sha af9a00deb21b5d795cabfeaa8d9060410377686d from https://github.com/mig4/setup-bats/releases/tag/v1.2.0
+        uses: mig4/setup-bats@af9a00deb21b5d795cabfeaa8d9060410377686d
+        with:
+          bats-version: 1.4.1      
+      - name: Setup Kind
+        # pinning to the sha aa272fe2a7309878ffc2a81c56cfe3ef108ae7d0 from https://github.com/engineerd/setup-kind/releases/tag/v0.5.0
+        uses: engineerd/setup-kind@aa272fe2a7309878ffc2a81c56cfe3ef108ae7d0
+        with:
+          version: "v0.11.1"
+      - name: Test
+        run: |
+          unset CI # Actions sets this var by default. We need to explicitly unset it so that build commit hash is not appended to driver image tag.
+          REGISTRY=${{ github.event.inputs.registry }} IMAGE_NAME=${{ github.event.inputs.driverImage }} IMAGE_VERSION=${{ github.event.inputs.driverImageVersion }} make e2e-helm-deploy e2e-mock-provider-container e2e-provider-deploy e2e-provider
```

Comment:
Could we set `REGISTRY`, `IMAGE_NAME`, `IMAGE_VERSION` as env var using `env:`?

Reply:
There is not `env:` like ADO. We can [set up env var](https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable), but we need to eventually access them like `${{ env.<NAME> }}`

Reply:
There is an `env` in github actions: https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-setting-an-environment-variable.

The input variables can be referenced as `${{ github.event.inputs.registry }}`

So overall this would be:

```yaml
run: <step>
env:
  REGISTRY: ${{ github.event.inputs.registry }}
  ...
```

Reply:
My bad. updated with env. Also tested with [v0.3.0](https://github.com/nilekhc/secrets-store-csi-driver/actions/runs/1214307274) and [v0.2.0](https://github.com/nilekhc/secrets-store-csi-driver/actions/runs/1214296506) to validate env var change.